src/credentials/CHIPCert.cpp

```diff
@@ -1145,6 +1145,38 @@ DLL_EXPORT CHIP_ERROR ChipEpochToASN1Time(uint32_t epochTime, chip::ASN1::ASN1Un
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ValidateChipRCAC(const ByteSpan & rcac)
+{
+    ChipCertificateSet certSet;
+    ChipCertificateData certData;
+    ValidationContext validContext;
+    uint8_t certType;
+
+    // Note that this function doesn't check RCAC NotBefore / NotAfter time validity.
+    // It is assumed that RCAC should be valid at the time of installation by definition.
+
+    ReturnErrorOnFailure(certSet.Init(&certData, 1));
+
+    ReturnErrorOnFailure(certSet.LoadCert(rcac, CertDecodeFlags::kGenerateTBSHash));
+
+    ReturnErrorOnFailure(certData.mSubjectDN.GetCertType(certType));
+    VerifyOrReturnError(certType == kCertType_Root, CHIP_ERROR_WRONG_CERT_TYPE);
+
+    VerifyOrReturnError(certData.mSubjectDN.IsEqual(certData.mIssuerDN), CHIP_ERROR_WRONG_CERT_TYPE);
```

Comment:
- The subject DN SHALL NOT encode any `matter-node-id` attribute.
  - The subject DN MAY encode at most one `matter-fabric-id` attribute.
    - If present, the `matter-fabric-id` attribute's value SHALL NOT be 0 (see <<ref_FabricReference>>).
- The subject DN SHALL NOT encode any `matter-icac-id` attribute.
- The subject DN MAY encode at most one `matter-rcac-id` attribute.
- The subject DN SHALL NOT encode any `matter-noc-cat` attribute.

^^^ Can all these be checked?

Reply:
All these checks are performed during `GetCertType()` call. I added `matter-noc-cat` attribute check to the `GetCertType()` method - that was missing.